# HG changeset patch
# Parent  abbddc13776b2a1fc8b301815207a0ffb04e17e0
Fixed two 32/64-bit compiler warnings in libmaxminddb v1.2.0

diff -r abbddc13776b -r 90fcb99d1394 src/maxminddb.c
--- a/src/maxminddb.c	Sun Aug 21 21:16:40 2016 -0400
+++ b/src/maxminddb.c	Sun Aug 21 21:22:45 2016 -0400
@@ -405,7 +405,7 @@ LOCAL const uint8_t *find_metadata(const
                           METADATA_MARKER, marker_len);
 
         if (NULL != tmp) {
-            max_size -= tmp - search_area;
+            max_size -= (ssize_t) (tmp - search_area);
             search_area = tmp;
 
             /* Continue searching just after the marker we just read, in case
@@ -1115,7 +1115,7 @@ int MMDB_read_node(MMDB_s *const mmdb, u
 LOCAL uint32_t data_section_offset_for_record(MMDB_s *const mmdb,
                                               uint64_t record)
 {
-    return record - mmdb->metadata.node_count - MMDB_DATA_SECTION_SEPARATOR;
+    return (uint32_t) (record - mmdb->metadata.node_count - MMDB_DATA_SECTION_SEPARATOR);
 }
 
 int MMDB_get_value(MMDB_entry_s *const start,
